extensions/kafka-connect/src/main/java/com/hazelcast/jet/kafka/connect/impl/ReadKafkaConnectP.java

```diff
@@ -88,8 +99,12 @@ public boolean complete() {
             localKafkaConnectStats.incrementSourceRecordPoll(sourceRecords.size());
             this.traverser = traverser(sourceRecords)
                     .map(rec -> {
+                        T projectedRecord = projectionFn.apply((SourceRecord) rec);
                         taskRunner.commitRecord((SourceRecord) rec);
-                        return rec;
+                        if (projectedRecord == null) {
+                            return Traversers.empty();
+                        }
+                        return projectedRecord;
```

Comment:
`Traverser#map` already supports filtering out null items, so we should be safe to return just `projectedRecord`, null or not.

Also I don't think you could return `Traversers.empty();` here, you would need `flatMap` for that.

Reply:
There was a bug here regarding native timestamps: https://github.com/hazelcast/hazelcast/pull/24203/commits/71728835c16e925f6f698793a05c7f5450568e2c

Also you were right about filtering records: https://github.com/hazelcast/hazelcast/pull/24203/commits/a815eb1f97c01ad1c703c6637c8871553096c392